### PR TITLE
[expo-sharing] Fixed sharing external URIs on Android

### LINKED ISSUE #9222: [expo-sharing] Sharing photos from MediaLibrary throws on Android
### Summary of Issue
When I try to share with `Sharing.shareAsync()` a photo from camera roll retrieved using `MediaLibrary.getAssetsAsync()` and `MediaLibrary.getAssetInfoAsync()`, iOS works great, but on Android I get an exception:
```
Failed to share the file: Failed to find configured root that contains /storage/emulated/0/Download/IMG_2704.PNG
```

I suspect it may be a problem of missing permissions in manifest.

### Environment
Android - emulator, API level 29 & 30
```
Expo CLI 3.22.1 environment info:
    System:
      OS: macOS 10.15.5
      Shell: 5.7.1 - /bin/zsh
    Binaries:
      Node: 14.5.0 - /usr/local/bin/node
      Yarn: 1.22.4 - /usr/local/bin/yarn
      npm: 6.14.5 - /usr/local/bin/npm
    IDEs:
      Android Studio: 4.0 AI-193.6911.18.40.6514223
      Xcode: 11.5/11E608c - /usr/bin/xcodebuild
    npmPackages:
      expo: ~38.0.8 => 38.0.8 
      react: ~16.11.0 => 16.11.0 
      react-dom: ~16.11.0 => 16.11.0 
      react-native: ~0.62.2 => 0.62.2 
      react-native-web: ~0.11.7 => 0.11.7 
```

### Reproducible Demo
Run [this demo from Snack](https://snack.expo.io/@barthap10/android-sharing-problem) on Android.

### Steps to Reproduce
Run demo from above. Ensure that you have at least one image in your Photos app (visible by `MediaLibrary.getAssetsAsync()`.

### Expected Behavior vs Actual Behavior
**Expected** - Sharing activity is opened and photo from Media Library can be shared
**Actual** - An exception (see summary) is shown.

### PR DESCRIPTION
**Pull request by barthap**
# Why
Fixes #9222.

# How
According to [Android docs](https://developer.android.com/reference/androidx/core/content/FileProvider#SpecifyFiles), `FileProvider` should have added `external-files` path in order to be able to access locations like `DCIM` or `Downloads`.
I added that line to `[sharing_provider_paths.xml](https://github.com/expo/expo/blob/master/packages/expo-sharing/android/src/main/res/xml/sharing_provider_paths.xml)`.

That fixed the issue, but after that I encountered following exception in Android Studio only:
```
E/DatabaseUtils: Writing exception to parcel
    java.lang.SecurityException: Permission Denial: reading expo.modules.sharing.SharingFileProvider uri content://host.exp.exponent.SharingFileProvider/expo_external_files/Download/IMG_2704.PNG from pid=16627, uid=1000 requires the provider be exported, or grantUriPermission()
        at android.content.ContentProvider.enforceReadPermissionInner(ContentProvider.java:820)
        at android.content.ContentProvider$Transport.enforceReadPermission(ContentProvider.java:684)
        at android.content.ContentProvider$Transport.enforceFilePermission(ContentProvider.java:674)
        at android.content.ContentProvider$Transport.openTypedAssetFile(ContentProvider.java:548)
        at android.content.ContentProviderNative.onTransact(ContentProviderNative.java:327)
        at android.os.Binder.execTransactInternal(Binder.java:1154)
        at android.os.Binder.execTransact(Binder.java:1123)
```
However, photos were sharing correctly, but I managed to fix it. I [followed the docs](https://developer.android.com/reference/androidx/core/content/FileProvider#Permissions) and updated `Intent` creation in `SharingModule.createSharingIntent()` by replacing `putExtra()` with `setData()` which resolved exception above.

# Test Plan
I only tested my changes using demo code from issue on API 29 & 30.

Sources I used:
 - [Android `FileProvider `docs](https://developer.android.com/reference/androidx/core/content/FileProvider)
 - [Android `Intent` docs](https://developer.android.com/reference/android/content/Intent.html)
 - [This article](https://medium.com/@quiro91/sharing-files-through-intents-part-2-fixing-the-permissions-before-lollipop-ceb9bb0eec3a), however it made me a bit confused about `Intent.FLAG_GRANT_READ_URI_PERMISSION` flag

### CODE REVIEW

**Review by lukmccall — CHANGES_REQUESTED** [review]
Please, don't forget about `CHANGELOG` ;)

File: packages/expo-sharing/android/src/main/res/xml/sharing_provider_paths.xml [hunk 1]
```diff
@@ -1,5 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <paths>
+  <external-path name="expo_external_files" path="." />a
```
Comment (lukmccall) [inline packages/expo-sharing/android/src/main/res/xml/sharing_provider_paths.xml:3]:
```suggestion
  <external-path name="expo_external_files" path="." />
```
🧹

**Review by lukmccall — APPROVED** [review]
LGTM 🎉 
Hovewer, I'm not sure if we want to add `<external-path name="expo_external_files" path="." />` to manifest. cc @mczernek

**Review by mczernek — APPROVED** [review]
Looks good to me :)

**Pull request merged**